### PR TITLE
Add RTL support to GridView

### LINKED ISSUE #11855: GridView RTL
For example SliverGridRegularTileLayout has an LTR bias.

### PR DESCRIPTION
**Pull request by abarth**
Fixes #11855

### CODE REVIEW

**Comment by abarth** [comment]
That was much more work than I expected.  😸

File: packages/flutter/lib/src/rendering/sliver.dart [hunk 1]
```diff
@@ -89,6 +89,21 @@ Axis axisDirectionToAxis(AxisDirection axisDirection) {
...
+/// [AxisDirection.right] for [TextDirection.ltr].
+AxisDirection textDirectionToAxisDirection(TextDirection textDirection) {
+  assert(textDirection != null);
+  switch (textDirection) {
+    case TextDirection.rtl:
+      return AxisDirection.left;
+    case TextDirection.ltr:
+      return AxisDirection.right;
+  }
+  return null;
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/sliver.dart]: @lukechurch This is the kind of thing for which I really wish I could put methods on enums.

File: packages/flutter/lib/src/widgets/viewport.dart [hunk 2]
```diff
@@ -213,6 +246,17 @@ class ShrinkWrappingViewport extends MultiChildRenderObjectWidget {
   /// bottom of the viewport.
   final AxisDirection axisDirection;
 
+  /// The direction in which child should be layed out in the cross axis.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/viewport.dart]: laid, not layed

**Comment by Hixie** [comment]
LGTM

**Pull request merged**